src/tree.rs

```diff
@@ -444,6 +444,49 @@ impl<T> Tree<T> {
         Result::Ok(())
     }
 
+    pub fn replace(&mut self, node_id: &NodeId, mut new_node: Node<T>) -> Result<Node<T>, NodeIdError>
+    {
+        use std::mem;
+
+        let (is_valid, error) = self.is_valid_node_id(node_id);
+        if !is_valid {
+            return Result::Err(error.expect("Tree::replace: Missing an error value on finding an invalid NodeId."));
+        }
+
+        // lets cheat a bit and return new_node ;)
+        // TODO, while this is elegant, this worksaround NodeBuilder's Parameters
+        // Should we drain the old children array instead?
+        mem::swap(self.get_mut_unsafe(node_id).data_mut(), new_node.data_mut());
```

Comment:
So, this brings up an interesting question: why would people want to call replace?

Now that I think more about it, all we can really do is what you've done above because:
- we can't mess with the children/parent of the existing `Node` since it wouldn't really be a "replace" at that point.
- the `Node` that we return will need to have its `parent` and `child` cleared anyway to prevent `NodeId` leakage.

So it really all boils down to just swapping the data from one `Node` with the data from the other (unless I'm missing something here).

If that's the case we may want to reassess what this method is actually intended to be used for and possibly change it's name to reflect that a bit better (for example `replace_node_data(&mut self, id: &NodeId, data: T) -> T` might be more clear).

Thoughts?